### PR TITLE
Perform runtime check for SSE 4.2 instructions

### PR DESCRIPTION
**Pull request by stanhu**
If SSE 4.2 is not available on the system, don't attempt to use SIMD instructions.

Relates to https://github.com/ohler55/oj/issues/789

### CODE REVIEW

File: ext/oj/parse.c [hunk 1]
```diff
@@ -213,6 +213,25 @@ static inline const char *scan_string_SIMD(const char *str, const char *end) {
 }
 #endif
 
+static bool cpu_supports_sse42() {
+#if __has_builtin(__builtin_cpu_supports)
```
Comment (stanhu) [inline ext/oj/parse.c:217]: Something's not working here. Hang on.

**Comment by ohler55** [comment]
Detecting earlier is better. Running the check on each parse is expensive. I'd rather the test was done when the Makefile is generated. There is a check in the current extconf.rb. Is that not working? If not I could use some help getting that to work on some of the older machines you might have access to. Is that possible?

**Comment by stanhu** [comment]
> Detecting earlier is better. Running the check on each parse is expensive. 

Doesn't this change only do the detection when the module is initialized?

> I'd rather the test was done when the Makefile is generated. There is a check in the current extconf.rb. Is that not working?

We package a build using SSE 4.2 platform, but some users run the code on older hardware. We either need a way to disable the SSE instructions outright, or make the check dynamic.

**Comment by ohler55** [comment]
I see, looks like what you have should be good then. I'll merge once tests are done and cut a release.

**Review by ohler55 — APPROVED** [review]

**Comment by ohler55** [comment]
Released

**Pull request merged**